airflow/jobs/scheduler_job.py

```diff
@@ -1298,7 +1324,11 @@ def _schedule_dag_run(
 
         self._verify_integrity_if_dag_changed(dag_run=dag_run, session=session)
         # TODO[HA]: Rename update_state -> schedule_dag_run, ?? something else?
-        schedulable_tis, callback_to_run = dag_run.update_state(session=session, execute_callbacks=False)
+        schedulable_tis, callback_to_run = dag_run.update_state(
+            session=session,
+            execute_callbacks=False,
+            notification=self.notify_dagrun_state_changed if self.enabled_dagrun_listener else None,
```

Comment:
it seems cleaner to have a "null notification" that is the default than to do this if / else

Reply:
Done